robottelo/cli/factory.py

```diff
@@ -1540,3 +1545,279 @@ def make_template(options=None):
+        u'product': options.get('product', None),
+    })
+    if result.return_code != 0:
+        raise CLIFactoryError('Failed to synchronize repository')
+    # Create CV if needed and associate repo with it
+    if not options.get('content-view-id', None):
+        cv_id = make_content_view({u'organization-id': org_id})['id']
+    else:
+        cv_id = options.get('content-view-id', None)
+    result = ContentView.add_repository({
+        u'id': cv_id,
+        u'repository-id': rhel_repo['id'],
+        u'organization-id': org_id,
+    })
+    if result.return_code != 0:
+        raise CLIFactoryError('Failed to add repository to content view')
+    # Publish a new version of CV
+    result = ContentView.publish({u'id': cv_id})
+    if result.return_code != 0:
+        raise CLIFactoryError('Failed to publish new version of content view')
+    # Get the version id
+    result = ContentView.info({u'id': cv_id})
+    cvv = result.stdout['versions'][-1]
+    # Promote version1 to next env
+    result = ContentView.version_promote({
+        u'id': cvv['id'],
+        u'to-lifecycle-environment-id': env_id,
+        u'organization-id': org_id,
+    })
+    if result.return_code != 0:
+        raise CLIFactoryError('Failed to promote version to next environment')
+    # Create activation key if needed and associate content view with it
+    if not options.get('activationkey-id', None):
+        activationkey_id = make_activation_key({
+            u'content-view-id': cv_id,
+            u'lifecycle-environment-id': env_id,
+            u'organization-id': org_id,
+        })['id']
+    else:
+        activationkey_id = options.get('activationkey-id', None)
+        # Given activation key may have no (or different) CV associated.
+        # Associate activation key with CV just to be sure
+        result = ActivationKey.update({
+            u'id': activationkey_id,
+            u'organization-id': org_id,
+            u'content-view-id': cv_id,
+        })
+        if result.return_code != 0:
+            raise CLIFactoryError('Failed to associate activation-key '
+                                  'with CV')
+    # List the subscriptions in given org
+    result = Subscription.list(
+        {u'organization-id': org_id},
+        per_page=False
+    )
+    # Add subscription to activation-key
+    for subscription in result.stdout:
+        if subscription['name'] == DEFAULT_SUBSCRIPTION_NAME:
+            if int(subscription['quantity']) == 0:
+                raise CLIFactoryError(
+                    'All the subscriptions are already consumed'
+                )
+            result = ActivationKey.add_subscription({
+                u'id': activationkey_id,
+                u'subscription-id': subscription['id'],
+                u'quantity': 1,
+            })
+            if result.return_code != 0:
+                raise CLIFactoryError(
+                    'Failed to add subscription to activation key'
+                )
```

Comment:
Subscription part is redundant between setup_org_for_a_custom_repo and setup_org_for_a_rh_repo.  This can be a separate function which both the other functions can call.  But this can be done in next PR.